src/rust/collections/ring.rs

```diff
@@ -39,6 +39,14 @@ pub struct RingBuffer<T> {
     is_managed: bool,
 }
 
+//======================================================================================================================
+// Traits
+//======================================================================================================================
+
+pub trait Ring: Sized {
```

Comment:
I wasn't sure if this was correct when I put it in. Do we know the size ahead of time and is this ok?

Reply:
You did it correct, because this should be implemented by a concrete types, and in rust all concrete types have a size.